src/toil/cwl/cwltoil.py

```diff
@@ -582,14 +584,20 @@ def run(self, file_store):
             self.runtime_context.tmpdir or file_store.getLocalTempDir())
         outdir = os.path.join(file_store.getLocalTempDir(), "out")
         os.mkdir(outdir)
+        # Create per-job temporary output directories under the CWL workdir,
+        # which we also are using as our job store.
+        #
+        # TODO: Why don't we just use Toil's file_store.getLocalTempDir() under
+        # Toil's --workDir? Why do we have two notions of workdir?
         top_tmp_outdir = self.workdir or os.environ["TMPDIR"]
         tmp_outdir_prefix = os.path.join(
             _makeNestedTempDir(top=top_tmp_outdir, seed=outdir, levels=2),
             "out_tmpdir")
-        self.openTempDirs.append(top_tmp_outdir)
+        self.openTempDirs.append(tmp_outdir_prefix)
```

Comment:
@adamnovak  FYI, this may not actually work as the temporary directory is something like `...out_tmpdir<tmpname>` (e.g. `/var/lib/toil/tmpPbBjlT/3/a/out_tmpdiraitngE/`).
Ideally, we'd want to remove the top-level `out_tmpdiraitngE` directory, but I'm not sure how we can get that as the suffix is generated by CWL. We also don't want to delete the upstream directory as it can be used by other tasks on the same worker.
My workaround so far has been to use `cwl_executor.output_dir` variable and remove those. It works, but it still leaves the temporary directory paths on the workers (I think that's fine though as they don't actually take up any space). See https://github.com/DataBiosphere/toil/commit/e149d827e9b07febf64d3bb9a2242165b25863fb for reference.

Reply:
The tmp_outdir_prefix is job-specific, right? And it is no longer needed at the end of the job. I can just add another level of subdirectory and delete the parent of whatever we pass along to CWL.

Reply:
@adamnovak  Your idea works too! However, `_makeNestedTempDir` doesn't actually use `mkdtemp`, so collisions are possible and likely as it's just based on the first two characters of a hash. Could you please add another layer that uses `mkdtemp` to ensure folders are indeed unique?